### DISCUSSION TITLE
hy2服务端是否支持配置独立的日志文件

### DISCUSSION THREAD

**Discussion by zhufeng** [apernet/hysteria#1027] [Q&A] [answered]
翻了下文档，关于hy2服务端的日志配置，只有主程序的启动参数的-l [日志级别]以及-f [console/json]，似乎没有找到哪里有配置可以把服务端的日志写入到单独的日志文件中？

是否考虑在config.yaml里面定义日志相关的配置呢，级别、格式、文件路径等。

**Comment by haruue** [comment, marked as answer]
如果你用手动启动服务端， 可以试试重定向， 例如 `hysteria server >> /path/to/logfile.txt 2>&1` 或者 `hysteria server 2>&1 | tee -a /path/to/logfile.txt`

如果你用 systemd 启动服务端， 可以试试 `systemctl edit hysteria-server.service` 并按照提示添加
```ini
StandardOutput=append:/path/to/logfile.txt
StandardError=append:/path/to/logfile.txt
```

**Reply by zhufeng** [reply]
好的，谢谢。

另外还想请问hy2服务端的防火墙配置的正确姿势应该是怎么样的。

我的环境是alma linux9使用自带的firewalld，firewalld的规则里面只放行https即443端口时，测试客户端无法连接。
后面自定义了一个规则，放行tcp+udp的40000-50000端口，似乎是可用了。
翻了文档似乎没有提到有关于hy2服务端的防火墙使用端口的说明。

**Reply by haruue** [reply]
https 的 443 端口是 TCP 的 443 端口， 而 Hysteria 需要的是 UDP 端口。 当服务端配置中包含 `listen: :443` 时， 你需要开放 UDP 443 端口。

要配置 firewalld 来开放指定的 UDP 端口， 请参阅此 Stack Exchange 回答： https://superuser.com/a/1455548

> 根据上面这个回答， 要开放 UDP 443 端口， 你可以试试执行以下命令。
>
> ```
> firewall-cmd --permanent --add-port=443/udp
> systemctl restart firewalld
> ```

以上关于 firewalld 的帮助仅供参考， 我不使用 ufw 和 firewalld 这类工具， 也没有它们的测试环境。

**Comment by zhufeng** [comment]
> https 的 443 端口是 TCP 的 443 端口， 而 Hysteria 需要的是 UDP 端口。 当服务端配置中包含 `listen: :443` 时， 你需要开放 UDP 443 端口。
> 
> 要配置 firewalld 来开放指定的 UDP 端口， 请参阅此 Stack Exchange 回答： https://superuser.com/a/1455548
> 
> > 根据上面这个回答， 要开放 UDP 443 端口， 你可以试试执行以下命令。
> > ```
> > firewall-cmd --permanent --add-port=443/udp
> > systemctl restart firewalld
> > ```
> 
> 以上关于 firewalld 的帮助仅供参考， 我不使用 ufw 和 firewalld 这类工具， 也没有它们的测试环境。

感谢，确实是协议问题，自带的https规则未包含udp的443，手动添加一个规则包含udp 443之后就可以了。
